File: LEET-CODE/01.twoSum.js
// Input: nums = [2,7,11,15], target = 9
// Output: [0,1]
// Explanation: Because nums[0] + nums[1] == 9, we return [0, 1]

// Input: nums = [3,2,4], target = 6
// Output: [1,2]

const twoSum = (nums, target) => {
  for (let i = 0; i < nums.length; i++) {
    for (let j = 0; j < nums.length; j++) {
      if (i != j) {
        if (nums[i] + nums[j] == target) {
          return [i, j];
        }
      }
    }
  }
};

File: 04.LEET-CODE/01.two_sum.cpp
// https://leetcode.com/problems/two-sum/
#include <iostream>
#include <vector>

using namespace std;

vector<int> twoSum(vector<int> &nums, int target)
{
	for (int i = 0; i < sizeof(nums); i++)
	{
		for (int j = 0; j < sizeof(nums); j++)
		{
			if (i != j)
			{
				if (nums[i] + nums[j] == target)
				{
					return {i, j};
				};
			};
		};
	};
	return {}; // why is this important
};

int main()
{
	vector<int> my_nums{2, 7, 11, 15};

	vector<int> arr = twoSum(my_nums, 9);

	for (int i = 0; i < arr.size(); i++)
	{
		cout << arr[i];
	}
	cout << endl;

	return 0;
};
